panel/src/mixins/preserveListeners.js

```diff
@@ -0,0 +1,43 @@
+import { wrap } from "@/helpers/array.js";
+
+/**
+ * Ensures that even when a component prohibits to
+ * inherit non-prop attributes, applied listeners
+ * are still preserved and attached to the root element
+ */
+export default {
+	data() {
+		return {
+			__listenersController: new AbortController()
+		};
+	},
+	mounted() {
+		// only if the component prohibits to inherit non-prop attributes
+		// we need to manually attach the listeners to the root element
+		if (this.$options.inheritAttrs === false) {
+			for (const attr in this.$attrs) {
+				// check if the attribute is an event listener
+				if (attr.startsWith("on") === true) {
+					// extract the event name
+					const event = attr.slice(2).toLowerCase();
+
+					// as there can be multiple listeners for the same event
+					// ensure that we can work with an array of listeners
+					const listeners = wrap(this.$attrs[attr]);
+
+					// attach each listener to the root element
+					for (const listener of listeners) {
+						this.$el.addEventListener(event, listener, {
+							// use an abort signal to ensure that all listeners
+							// are removed when the component is destroyed
+							signal: this.__listenersController.signal
+						});
+					}
+				}
+			}
+		}
+	},
+	destroyed() {
+		this.__listenersController.abort();
+	}
+};
```

Comment:
`Discussion:` Unsure if this could have an impact on performance. But so far haven't found another way to still use listeners even when using `inheritAttrs: false`.

Reply:
@johannschopplich @bastianallgeier This is one part of the Vue 3 PR where additional brainpower (and experience) could be helpful. This seems a bit ugly but I did not fine any other solution, maybe you know one. The issue is that in Vue 3 when we set `inheritAttrs: false` also listeners get ignored and are not attached to the root element of the component. This mixin that gets applied to every component adds them manually, which seems to work but also seems like a hack.

Reply:
I haven't had this issue with Vue before. May feel hacky, but yeah, pretty solid solution.

Reply:
Side note: Isn't `destroyed` replaced by `unmounted` in Vue 3?